cluster-test/test-orchestrator/test_yaml_parser.go

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+
+	"gopkg.in/yaml.v2"
+)
+
+// testYaml is used to unmarshal test declared in the yaml files.
+type testYaml struct {
+	Config    configYaml
+	Scenarios []*scenarioYaml
+}
+
+type configYaml struct {
+	// TODO(wieger): define
+}
+
+// scenarioYaml captures the information of a scenario.
+type scenarioYaml struct {
+	Name string
+	Size string
+	Desc string
+
+	Script  []map[string]string
+	Measure []string
+	Runs    [][]string
+}
+
+func parse(bts []byte) (scns []*Scenario, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			scns = nil
+			err = errors.New(fmt.Sprint(r))
+		}
+	}()
+
+	return parseScenarios(bts), nil
+}
+
+func parseScenarios(bts []byte) []*Scenario {
+	testYaml := &testYaml{}
+	err := yaml.Unmarshal([]byte(bts), testYaml)
+	if err != nil {
+		panic("failed to unmarshal scenario yaml")
+	}
+
+	return extractScenarios(testYaml)
+}
+
+// extractScenarios returns a scenario for every element in the runs list.
+func extractScenarios(runs *testYaml) []*Scenario {
+	var result []*Scenario
+	for _, scenarioData := range runs.Scenarios {
+		for i := 1; i < len(scenarioData.Runs); i++ {
```

Comment:
You'r missing one iteration here because you're looping from 1 till the number of runs.